x-pack/plugins/observability_solution/investigate_app/public/pages/details/components/investigation_details/index.tsx

```diff
@@ -10,23 +10,41 @@ import type { InvestigateWidgetCreate } from '@kbn/investigate-plugin/public';
 import { AuthenticatedUser } from '@kbn/security-plugin/common';
 import { keyBy, noop } from 'lodash';
 import React, { useEffect, useMemo, useRef } from 'react';
+import { useParams } from 'react-router';
 import useAsync from 'react-use/lib/useAsync';
 import { AddObservationUI } from '../../../../components/add_observation_ui';
 import { InvestigateSearchBar } from '../../../../components/investigate_search_bar';
 import { InvestigateWidgetGrid } from '../../../../components/investigate_widget_grid';
+import { useAddInvestigationNote } from '../../../../hooks/use_add_investigation_note';
 import { useDateRange } from '../../../../hooks/use_date_range';
+import { useFetchInvestigation } from '../../../../hooks/use_fetch_investigation';
 import { useKibana } from '../../../../hooks/use_kibana';
+import { InvestigationDetailsPathParams } from '../../types';
 import { InvestigationNotes } from '../investigation_notes/investigation_notes';
 
-function InvestigationDetailsWithUser({ user }: { user: AuthenticatedUser }) {
+function InvestigationDetailsWithUser({
+  user,
+  investigationId,
+}: {
+  user: AuthenticatedUser;
+  investigationId: string;
+}) {
   const {
     dependencies: {
       start: { investigate },
     },
   } = useKibana();
-  const widgetDefinitions = useMemo(() => investigate.getWidgetDefinitions(), [investigate]);
+  const widgetDefinitions = investigate.getWidgetDefinitions();
   const [range, setRange] = useDateRange();
 
+  const { data: investigationData } = useFetchInvestigation({ id: investigationId });
+  const { mutateAsync: addInvestigationNote } = useAddInvestigationNote();
+  const handleAddInvestigationNote = async (note: string) => {
+    await addInvestigationNote({ investigationId, note: { content: note } });
+    // todo: move following to mutate hook
+    await addNote(note);
+  };
```

Comment:
This is obviously not good, but i feel we need to refactor the useInvestigation hook (remove it?) to sync what we get from the API.

Reply:
do we need to maintain internal store? If we were allowing users to manually save the investigation, we would need it. But since we are auto-updating the investigation with API, I can't think of a case where we would need to maintain internal store. Maybe I am missing some use case where we will use this?